pkg/controller/plan/kubevirt.go

```diff
@@ -2686,21 +2690,38 @@ func (r *VirtualMachine) Conditions() (cnd *libcnd.Conditions) {
 	return
 }
 
-// Create an OwnerReference from a VM.
-func vmOwnerReference(vm *cnv.VirtualMachine) (ref meta.OwnerReference) {
+// Generic function to create OwnerReferences based on the input type.
+func setOwnerReferences[T any](obj T) []meta.OwnerReference {
+	var apiVersion, kind string
+	var name string
+	var uid types.UID
 	blockOwnerDeletion := true
```

Comment:
As discussed in mtg (Apr 10) we do not want to block in case of Plan owner, to unblock deletion of plans stuck on children objects deletion, stuck plans may block finalize call of children creating cyclic block loop.